src/worker_utils.ts

```diff
@@ -0,0 +1,62 @@
+import { Pool, PoolClient } from "pg";
+import { WorkerUtilsOptions, TaskOptions, Job } from "./interfaces";
+import { makeWithPgClientFromPool, makeAddJob } from "./helpers";
+import { migrate } from "./migrate";
+import { defaultLogger } from "./logger";
+import { withReleasers, assertPool } from "./runner";
+
+const processPublisherOptions = async (options: WorkerUtilsOptions) => {
+  const { logger = defaultLogger } = options;
+  return withReleasers(async (releasers, _) => {
+    const pgPool: Pool = await assertPool(options, releasers, logger);
+
+    const withPgClient = makeWithPgClientFromPool(pgPool);
+
+    // Migrate
+    await withPgClient(client => migrate(client));
+
+    return { pgPool, withPgClient };
+  });
+};
+
+export class WorkerUtils {
+  private dbTools: Promise<{
+    withPgClient: <T>(
+      callback: (pgClient: PoolClient) => Promise<T>
+    ) => Promise<T>;
+    pgPool: Pool;
+  }>;
+
+  constructor(options: WorkerUtilsOptions) {
+    this.dbTools = processPublisherOptions(options);
+  }
+
+  public async addJob(
+    identifier: string,
+    payload: any = {},
+    options: TaskOptions = {}
+  ): Promise<Job> {
+    const { withPgClient } = await this.dbTools;
+    const addJob = makeAddJob(withPgClient);
+    return addJob(identifier, payload, options);
+  }
+
+  async end(): Promise<void> {
+    const { pgPool } = await this.dbTools;
+    await pgPool.end();
```

Comment:
This is not the right thing to do - we might not own the `pgPool`, so ending it could lead to very unexpected results for the user. This is where the `dbTools.release()` method would be called, which handles this complexity for you.

```suggestion
    (await this.dbTools).release();
```